lib/stdlib/test/calendar_SUITE.erl

```diff
@@ -81,6 +83,79 @@ big_gregorian_days(Config) when is_list(Config) ->
     MaxDays = calendar:date_to_gregorian_days({?BIG_END_YEAR, 1, 1}),
     check_gregorian_days(Days, MaxDays).
 
+%% Tests edge cases for the Neri-Schneider algorithm.
+%% This includes epoch boundaries, leap years, century boundaries,
+%% and 400-year era boundaries.
+gregorian_days_edge_cases(Config) when is_list(Config) ->
+    %% Test epoch (day 0 = Jan 1, year 0)
+    0 = calendar:date_to_gregorian_days(0, 1, 1),
+    {0, 1, 1} = calendar:gregorian_days_to_date(0),
+
+    %% Test year 0 boundaries (year 0 is a leap year)
```

Comment:
Now _that_ got me thinking 🤔 Turns out [there is no year `0` in the Gregorian calendar](https://en.wikipedia.org/wiki/Year_zero), year 1BC is immediately followed by year 1AD 🤯

And while [the `calendar` module doc](https://www.erlang.org/doc/apps/stdlib/calendar.html) states that the Gregorian calendar of this module "is extended back to year 0", IMO this simply doesn't work out. There _is_ no year `0` in the Gregorian calendar, period. Year `0`, if you would instead assume the _astronomical_ calendar, would map to year 1BC in the Gregorian calendar. In any case, this would mean that "year `0`" (1BC) is _not_ a leap year.

@IngelaAndin (and maybe also @bjorng), what are your thoughts about this?